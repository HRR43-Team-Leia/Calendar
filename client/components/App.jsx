import React, { Component } from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import Calendar from './Calendar.jsx';

const CalendarArea = styled.div`
  position: relative;
  height: 450px;
  width: 800px;
  overflow: hidden;
  background-color: wheat;
`;

const Header = styled.div`
  color: blue;
`;

// how to have one element on the top and another at the bottom?
const Btn = styled.button`
  align-items: flex-end;
  color: blue;
`;

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      months: [],
      expId: props.expId,
    };
  }

  componentDidMount() {
    const { expId } = this.state;
    axios.get(`http://localhost:3005/calendar/${expId}`)
      .then((res) => {
        this.setState({
          months: res.data[0].dates,
        });
      })
      .catch((err) => {
        console.log('ERROR from axios get request: ', err);
      });
  }

  render() {
    const { months } = this.state;
    console.log(months);
    return (
      <CalendarArea>
        <Header>Availability</Header>
        <Calendar months={months} />
        <Btn type="submit">select a date</Btn>
      </CalendarArea>
    );
  }
}

App.propTypes = {
  expId: PropTypes.number,
};
App.defaultProps = {
  expId: 1,
};



export default App;
