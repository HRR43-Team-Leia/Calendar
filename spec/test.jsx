//'use strict';

import React from 'react';
import App from '../client';
import renderer from 'react-test-renderer';

const puppeteer = require('puppeteer');
const pageUrl = 'http://localhost:3005/';


let page;
let browser;
const width = 800;
const height = 600;

beforeAll (async () => {
  browser = await puppeteer.launch({
    headless:false,
    slowMo:80,
    args: [`--window-size=$(width), $(height)`]
  });
  page = await browser.newPage();
  await page.setViewport({width, height});
});

describe('TESTS', () => {

  beforeEach(async () => {
    await page.goto(pageUrl, {waitUntil: 'networkidle2'});
  })

  test('first test', async () => {
    var div = 'h1'
    const title = await page.$eval(div, e => e.textContent);
    expect(title).toEqual('React Works!')
  });

  test('renderds something', async ()=> {
    const tree = renderer
      .create( <App /> )
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
});

afterAll (() => {
  browser.close();
})