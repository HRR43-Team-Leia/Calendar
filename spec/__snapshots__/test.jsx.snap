// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTS renderds something 1`] = `
<div>
  <h1>
    React Works!
  </h1>
  <table>
    <tr>
      <td />
      <td>
        1
      </td>
      <td>
        2
      </td>
      <td>
        3
      </td>
    </tr>
    <tr>
      <td>
        3
      </td>
      <td />
      <td>
        6
      </td>
      <td>
        7
      </td>
    </tr>
  </table>
  <input
    id="start"
    max="2018-12-31"
    min="2018-01-01"
    name="trip-start"
    type="date"
    value="2018-07-22"
  />
</div>
`;
